server/resp/src/main/java/org/infinispan/server/resp/commands/cluster/KEYSLOT.java

```diff
@@ -38,7 +40,17 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, Channel
       KeyPartitioner partitioner = SecurityActions.getCacheComponentRegistry(respCache)
             .getComponent(KeyPartitioner.class);
 
-      CompletionStage<Integer> cs = CompletableFuture.completedFuture(partitioner.getSegment(arguments.get(1)));
+      CompletionStage<Integer> cs;
+      if (partitioner instanceof CRC16HashFunctionPartitioner) {
+         Hash hash = ((CRC16HashFunctionPartitioner) partitioner).getHash();
+         byte[] key = arguments.get(1);
+         int h = hash.hash(key);
+         int s = partitioner.getSegment(key);
```

Comment:
Don't we require the partitioner to use CRC16 always? I think we should probably add methods to the KeyPartitioner like `hash(Object obj)` and `getSegmentFromHash(int hash)` so we can reuse it without having to calculate the hash twice.